VMEncryption/main/EncryptionSettingsUtil.py

```diff
@@ -143,32 +172,54 @@ def dict_to_name_value_array(values):
                         })
             return array
 
+        # Get disk data from disk_util
+        # We get a list of tuples i.e. [(scsi_controller_id, lun_number),.]
+        data_disk_controller_ids_and_luns = disk_util.get_azure_data_disk_controller_and_lun_numbers(azure_vhd_dev_items)
+
+        def controller_id_and_lun_to_settings_data(scsi_controller, lun_number):
+            return {
+                "ControllerType": "SCSI",
+                "ControllerId": scsi_controller,
+                "SlotId": lun_number,
+                "Volumes": [{
+                    "VolumeType": "DataVolume",
+                    "ProtectorFileName": protector_name,
+                    "SecretTags": dict_to_name_value_array({
+                        "DiskEncryptionKeyFileName": CommonVariables.encryption_key_file_name + "_" + str(scsi_controller) + "_" + str(lun_number),
+                        "DiskEncryptionKeyEncryptionKeyURL": kek_url,
+                        "DiskEncryptionKeyEncryptionAlgorithm": kek_algorithm,
+                        "MachineName": machine_name})
+                    }]
+                }
+
+        data_disks_settings_data = [ controller_id_and_lun_to_settings_data(scsi_controller, lun_number)
+                                    for (scsi_controller, lun_number) in data_disk_controller_ids_and_luns]
+
+        if root_vhd_needs_stamping:
+            data_disks_settings_data.append({
+                "ControllerType": "IDE",
+                "ControllerId": 0,
+                "SlotId": 0,
+                "Volumes": [{
+                    "VolumeType": "OsVolume",
+                    "ProtectorFileName": protector_name,
+                    "SecretTags": dict_to_name_value_array({
+                        "DiskEncryptionKeyFileName": CommonVariables.encryption_key_file_name,
+                        "DiskEncryptionKeyEncryptionKeyURL": kek_url,
+                        "DiskEncryptionKeyEncryptionAlgorithm": kek_algorithm,
+                        "MachineName": machine_name})
+                    }]
+                })
+
         data = {
-            "DiskEncryptionDataVersion": "2.0",
+            "DiskEncryptionDataVersion": "3.0",
```

Comment:
2.0 is available in all regions, but I believe 3.0 rollout is still underway, maybe a comment here on availability would help

Reply:
will address in next update